src/gdsync.cpp

```diff
@@ -418,6 +418,12 @@ static int gds_fill_poll(gds_op_list_t &ops, CUdeviceptr ptr, uint32_t magic, in
         assert((((unsigned long)ptr) & 0x3) == 0);
 
         bool need_flush = (flags & GDS_WAIT_POST_FLUSH) ? true : false;
+
+//Flag CU_STREAM_WAIT_VALUE_FLUSH generates an error with CUDA 9.x
+#if defined(__x86_64__) || defined (__i386__)
+        need_flush=false;
+#endif
```

Comment:
can you print a warning only once? like
"RDMA consistency for pre-launched GPU work is not guaranteed at the moment"

Also, recording here that we would like to do the same for POWER8, though in that case peer memory mappings will generate an error.